File: solutions/0121-best-time-to-buy-and-sell-stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    if(prices.length==0){
        return 0;
    }
    let n=0;
    let minprice=Infinity;
    for(let i=0;i<prices.length;i++){
        if(prices[i]<minprice){
            minprice=prices[i]
        }
        else if(prices[i]-minprice>n){
            n=prices[i]-minprice;
        }
    }
 
    return n;
};

File: solutions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if(len(prices)==0):
            return 0
        profit=0
        minprice=float('inf')
        for i in range(len(prices)):
            if(prices[i]<minprice) :
                minprice= prices[i]

            
            elif(prices[i]-minprice > profit) :
                profit=prices[i]-minprice 
            

        return profit
